src/cloud-providers/ibmcloud/provider.go

```diff
@@ -177,6 +177,13 @@ func (p *ibmcloudVPCProvider) getInstancePrototype(instanceName, userData, insta
 			Enabled:  core.BoolPtr(true),
 			Protocol: core.StringPtr(vpcv1.InstanceMetadataServicePatchProtocolHTTPConst),
 		},
+		ConfidentialComputeMode: core.StringPtr("tdx"), // TODO when available: vpcv1.InstanceConfidentialComputeModeTdxConst
```

Comment:
Will the ConfidentialComputeMode be always tdx? No other TEEs (eg. snp) ?

Reply:
My understanding is at this point TDX is the only TEE we plan to support in IBM Cloud. In the future we could add another config parameter to support others, but unless I'm missing something, I don't think there would be a backward compatibility issue when we do that.

Reply:
I confirmed that TDX is all that IBM Cloud plans to support for a while, at least through 2025.